src/app/query/sbtc/sponsored-transactions.ts

```diff
@@ -0,0 +1,126 @@
+import { useEffect, useState } from 'react';
+
+import { bytesToHex } from '@stacks/common';
+import { StacksTransaction } from '@stacks/transactions';
+import axios from 'axios';
+
+import { FeeTypes, type Fees } from '@leather.io/models';
+import { type NextNonce } from '@leather.io/query';
+
+import { logger } from '@shared/logger';
+
+import { generateUnsignedTransaction } from '@app/common/transactions/stacks/generate-unsigned-txs';
+import { useCurrentStacksAccountAddress } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+
+interface TransactionBase {
+  options: any;
+  transaction: StacksTransaction | undefined;
+}
+
+interface SbtcSponsorshipVerificationResult {
+  isVerifying: boolean;
+  result: SbtcSponsorshipEligibility | undefined;
+}
```

Comment:
Could this be discriminating union to help types when consuming?

```suggestion
interface SbtcSponsorshipVerificationResultPending {
  isVerifying: false;
}
interface SbtcSponsorshipVerificationResultVerified {
  isVerifying: true;
  result: SbtcSponsorshipEligibility;
}
type SbtcSponsorshipVerificationResult = SbtcSponsorshipVerificationResultPending | SbtcSponsorshipVerificationResultVerified;
```